add channel handling scripts

szuecs opened the pull request:
add channel handling scripts

szuecs committed:
add channel handling scripts

Signed-off-by: Sandor Szücs <sandor.szuecs@zalando.de>

njuettner commented:
👍

aermakov-zalando commented:
Can you add a hashbang and `set -e`? This will happily continue on errors.

mikkeloscar commented on alpha-to-beta.sh:
```diff
@@ -0,0 +1,12 @@
+git co alpha
```
I think `co` is an alias? Please spell out the full command `checkout` :)

Same for `br` -> `branch` etc.

hjacobs committed:
full git comment, shebang

szuecs committed:
add PR handling if hub is installed and prepare message

szuecs committed:
prepare PR message and title and create pull request
please update your github.com/github/hub
known issue: on create-pullrequest it opens your editor https://github.com/github/hub/issues/1741

szuecs commented:
I added a bit more automation
* prepares PR message and title
* if "hub" is installed then it creates a PR with title and body and adds our testing label automatically

szuecs commented:
:+1:

aermakov-zalando commented:
Thanks!

Shellcheck prints some warnings, at least one of those looks that it matters:

    In alpha-to-beta.sh line 16:
    echo "alpha-to-beta\n"> $MSG
          ^-- SC2028: echo won't expand escape sequences. Consider printf.

The rest looks relatively benign, but I'd still fix the reported issues.

Additionally, the scripts only differ in the branch names. Would you consider having a `prepare-pr` script instead that would accept source and target branches as arguments?

szuecs committed:
add script params

szuecs committed:
rename script and drop the unecassary one

szuecs commented:
@aermakov-zalando changed as commented

szuecs committed:
use shellcheck to fix all warnings

szuecs commented:
:+1:

mikkeloscar commented:
:+1:
